--- jfsutils-1.1.12/libfs/devices.c.orig	2007-04-07 19:09:56.000000000 +0200
+++ jfsutils-1.1.12/libfs/devices.c	2008-04-29 10:27:20.447872421 +0200
@@ -15,6 +15,13 @@
  *   along with this program;  if not, write to the Free Software
  *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
+
+/*
+ * Override config.h by defining _LARGEFILE_SOURCE here
+ */
+#define _LARGEFILE_SOURCE
+
 #include <config.h>
 #include <errno.h>
 #include <fcntl.h>
