--- jfsutils-1.1.1.org/libfs/log_dump.c	Mon Sep 23 21:07:43 2002
+++ jfsutils-1.1.1/libfs/log_dump.c	Sun Mar 16 14:57:59 2003
@@ -22,6 +22,7 @@
 #include <memory.h>
 #include <string.h>
 #include <fcntl.h>
+#include <errno.h>
 #include "jfs_endian.h"
 #include "jfs_filsys.h"
 #include "jfs_superblock.h"
--- jfsutils-1.1.1.org/libfs/log_map.c	Mon Sep 23 21:07:43 2002
+++ jfsutils-1.1.1/libfs/log_map.c	Sun Mar 16 14:56:27 2003
@@ -25,6 +25,7 @@
 #include <stdlib.h>
 #include <memory.h>
 #include <string.h>
+#include <errno.h>
 #include "jfs_endian.h"
 #include "jfs_filsys.h"
 #include "jfs_superblock.h"
