--- jfsutils-1.1.15/fscklog/display.c.formatsec	2013-02-16 12:14:55.373045011 +0100
+++ jfsutils-1.1.15/fscklog/display.c	2013-02-16 12:15:17.429592187 +0100
@@ -182,7 +182,7 @@ void dump_service_log()
 				} else {
 					/* the record looks ok */
 					msg_txt = &log_entry[log_entry_pos];
-					printf(msg_txt);
+					fputs(msg_txt, stdout);
 					/*
 					 * set up for the next record
 					 */
--- jfsutils-1.1.15/fscklog/fscklog.c.formatsec	2013-02-16 12:14:10.863286141 +0100
+++ jfsutils-1.1.15/fscklog/fscklog.c	2013-02-16 12:14:47.609753735 +0100
@@ -252,8 +252,8 @@ int v_send_msg(int msg_num, const char *
 
 	sprintf(debug_detail, " [%s:%d]\n", basename(file_name), line_number);
 
-	printf(msg_string);
-	printf(debug_detail);
+	fputs(msg_string, stdout);
+	fputs(debug_detail, stdout);
 
 	return 0;
 }
--- jfsutils-1.1.15/logdump/helpers.c.formatsec	2013-02-16 12:21:02.697721709 +0100
+++ jfsutils-1.1.15/logdump/helpers.c	2013-02-16 12:21:25.810929827 +0100
@@ -95,8 +95,8 @@ int v_fsck_send_msg(int msg_num, const c
 
 	sprintf(debug_detail, " [%s:%d]\n", file_name, line_number);
 
-	printf(msg_string);
-	printf(debug_detail);
+	fputs(msg_string, stdout);
+	fputs(debug_detail, stdout);
 
 	return 0;
 }
